test/index.js

```diff
@@ -710,136 +628,124 @@ describe('Mongo', () => {
         });
 
-        it('passes an error to the callback when there is an error getting the collection', (done) => {
+        it('passes an error to the callback when there is an error getting the collection', async () => {
 
             const options = {
                 uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
 
-            mongo.start(() => {
-
-                mongo.getCollection('', (err, result) => {
-
-                    expect(err).to.exist();
-                    expect(result).to.not.exist();
-                    done();
-                });
-            });
+            await mongo.start();
+            try {
+                const result = await mongo.getCollection('');
+                expect(result).to.not.exist();
+            }
+            catch (err) {
+                expect(err).to.exist();
+            }
         });
 
-        it('passes an error to the callback when ensureIndex fails', (done) => {
+        it('passes an error to the callback when ensureIndex fails', async () => {
 
             const options = {
                 uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
 
-            mongo.start(() => {
+            await mongo.start();
 
-                mongo.db.collection = function (item, callback) {
+            mongo.db.collection = function (item) {
 
-                    return callback(null, {
-                        ensureIndex: function (fieldOrSpec, options2, callback2) {
+                return Promise.resolve({
+                    ensureIndex: function (fieldOrSpec, options2) {
 
-                            return callback2(new Error('test'));
-                        }
-                    });
-                };
-
-                mongo.getCollection('testcollection', (err, result) => {
-
-                    expect(err).to.exist();
-                    expect(result).to.not.exist();
-                    done();
+                        return Promise.reject(new Error('test'));
+                    }
                 });
-            });
+            };
+
+            try {
+                const result = await mongo.getCollection('testcollection');
+                expect(result).to.not.exist();
+            }
+            catch (err) {
+                expect(err).to.exist();
+            }
         });
     });
 
     describe('get()', () => {
 
-        it('passes an error to the callback when the connection is closed', (done) => {
+        it('returns an error when the connection is closed', async () => {
 
             const options = {
                 uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
 
-            mongo.get('test', (err) => {
-
+            try {
```

Comment:
I suspect that if mongo.get does not throw, it still would pass, because expects in catch would not get called. There is multiple tests like this. Not sure whats the best practice for these tests.

@Marsup Any hint? :-)

Reply:
Throw something different on the new line, or return in the catch and throw after.

Reply:
Alright, thank you for pointing out a solution 👍